### PR TITLE
changes to form submission metrics

### PR DESCRIPTION
**Pull request by snopoke**
* add domain to tags
* add metric for number of cases processed
* add metric for number of forms processed

Best read per commit :fish: 

FYI @benrudolph @dannyroberts

### CODE REVIEW

File: corehq/ex-submodules/casexml/apps/case/tests/test_bugs.py [hunk 1]
```diff
@@ -120,12 +120,12 @@ def test_duplicate_case_properties(self):
             CaseBlock(create=True, case_id=case_id).as_xml()
         ])
         xml_data = self.get_xml('duplicate_case_properties')
-        response, form, [case] = submit_form_locally(xml_data, 'test-domain')
-        self.assertEqual("", case.dynamic_case_properties()['foo'])
+        result = submit_form_locally(xml_data, 'test-domain')
+        self.assertEqual("", result.case.dynamic_case_properties()['foo'])
```
Comment (kaapstorm) [inline corehq/ex-submodules/casexml/apps/case/tests/test_bugs.py]: `result.cases` no?

Comment (snopoke) [inline corehq/ex-submodules/casexml/apps/case/tests/test_bugs.py]: no, I added a shortcut `.case` for when there's only one case

Comment (snopoke) [inline corehq/ex-submodules/casexml/apps/case/tests/test_bugs.py]: but thanks, good :eye:

**Review by kaapstorm — APPROVED** [review]

File: corehq/apps/cleanup/tests/test_fix_forms_and_apps_with_missing_xmlns.py [hunk 2]
```diff
@@ -51,9 +51,9 @@ def tearDownClass(cls):
```
Comment (benrudolph) [inline corehq/apps/cleanup/tests/test_fix_forms_and_apps_with_missing_xmlns.py]: hero for making this refactor

File: corehq/apps/receiverwrapper/views.py [hunk 3]
```diff
@@ -95,10 +95,12 @@ def _process_form(request, domain, app_id, user_id, authenticated,
 
     response = result.response
 
-    backend_tag = ('backend:sql' if should_use_sql_backend(domain) else
-                   'backend:couch')
     counter_name = 'commcare.xform_submissions.{}'.format(response.status_code)
```
Comment (benrudolph) [inline corehq/apps/receiverwrapper/views.py]: nit: might be more useful to have the status code as a tag as opposed to part of the metric

**Review by benrudolph — APPROVED** [review]

File: corehq/apps/receiverwrapper/views.py [hunk 4]
```diff
@@ -91,12 +90,15 @@ def _process_form(request, domain, app_id, user_id, authenticated,
...
-                   'backend:couch')
-    counter_name = 'commcare.xform_submissions.{}'.format(response.status_code)
-    datadog.statsd.increment(counter_name, tags=[backend_tag])
+    response = result.response
+
+    tags = [
+        'backend:sql' if should_use_sql_backend(domain) else 'backend:couch',
+        u'domain:{}'.format(domain)
+    ]
+    datadog_counter('commcare.xform_submissions.count', tags=tags + ['status_code:{}'.format(response.status_code)])
```
Comment (dimagimon) [inline corehq/apps/receiverwrapper/views.py]: E501 line too long (116 > 115 characters)

**Comment by benrudolph** [comment]
```
======================================================================
ERROR: test_globally_unique_form_id (corehq.form_processor.tests.test_basic_cases.FundamentalCaseTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/form_processor/tests/test_basic_cases.py", line 333, in test_globally_unique_form_id
    _, xform, _ = submit_form_locally(form.format(form_id=form_id), 'domain1')
ValueError: too many values to unpack
----------------------------------------------------------------------
```

**Pull request merged**